samples/peptide_mixture_grxmc_ideal.py

```diff
@@ -264,27 +263,41 @@
     err_xi_plus.append(err_concentration_plus/ionic_strength_res)
     print("pH = {:6.4g} done".format(pH_value))
    
-# Calculate the ideal titration curve of the peptide with Henderson-Hasselbach equation
-pH_range_HH = np.linspace(2, 12, num=100)
-HH_charge_dict = pmb.calculate_HH_Donnan(c_macro={peptide1: pep1_concentration, peptide2: pep2_concentration}, c_salt=c_salt, pH_list=pH_range_HH)
-Z_HH_Donnan = HH_charge_dict["charges_dict"]
-pH_sys = HH_charge_dict["pH_system_list"]
-xi = HH_charge_dict["partition_coefficients"]
-
-fig, ax = plt.subplots(figsize=(10, 7))
-plt.errorbar(pH_range, np.asarray(Z_pH)/N_peptide1_chains, yerr=np.asarray(err_Z_pH)/N_peptide1_chains, fmt = 'o', capsize=3, label='Simulation')
-ax.plot(pH_range_HH, np.asarray(Z_HH_Donnan[peptide1])+np.asarray(Z_HH_Donnan[peptide2]), "--r", label='HH+Donnan')
-plt.legend()
-plt.xlabel('pH')
-plt.ylabel('Charge of the peptide 1 + peptide 2 / e')
-plt.show()
-plt.close()
-
-fig, ax = plt.subplots(figsize=(10, 7))
-plt.errorbar(pH_range, xi_plus, yerr=err_xi_plus, fmt = 'o', capsize=3, label='Simulation')
-ax.plot(pH_range_HH, np.asarray(xi), "-k", label='HH+Donnan')
-plt.legend()
-plt.xlabel('pH')
-plt.ylabel(r'partition coefficient $\xi_+$')
-plt.show()
-plt.close()
+
+if args.mode == 'standard' or args.mode == 'unified':
```

Comment:
In general I would try to avoid to have too many "test-specific" features in the codes of the samples, because they are meant to be examples for our users. 
I would try to refactor this a bit as follows:
- drop the test specific modes
- instead add a boolean flag `plot` to control if the script does the plotting or not, so you can deactivate that part when running the test (all the rest is actually not test specific)
- save the data to some files that can be then be read by the test and check that both produce the same result